api/src/main/java/org/openmrs/module/queue/api/impl/QueueEntryServiceImpl.java

```diff
@@ -115,20 +127,34 @@ public String generateVisitQueueNumber(Location location, Queue queue, Visit vis
 		if (location == null || queue == null || visit == null || visitAttributeType == null) {
 			throw new APIException("Sufficient parameters not supplied for generation of VisitQueueNumber");
 		}
-		String queueNumber = dao.generateVisitQueueNumber(location, queue);
+		QueueEntrySearchCriteria criteria = new QueueEntrySearchCriteria();
+		criteria.setHasVisit(Boolean.TRUE);
+		criteria.setQueues(Collections.singletonList(queue));
+		criteria.setLocations(Collections.singletonList(location));
+		Date onOrAfter = Date.from(LocalDateTime.now().with(LocalTime.MIN).atZone(ZoneId.systemDefault()).toInstant());
+		criteria.setStartedOnOrAfter(onOrAfter);
+		Date onOrBefore = Date.from(LocalDateTime.now().with(LocalTime.MAX).atZone(ZoneId.systemDefault()).toInstant());
+		criteria.setStartedOnOrAfter(onOrBefore);
+		Long nextQueueNumber = getCountOfQueueEntries(criteria) + 1;
+		String paddedString = StringUtils.leftPad(String.valueOf(nextQueueNumber), 3, "0");
+		String serviceName = queue.getName().toUpperCase();
+		String prefix = serviceName.length() < 3 ? serviceName : serviceName.substring(0, 3);
+		String queueNumber = prefix + "-" + paddedString;
```

Comment:
@CynthiaKamau - see this section above.  This basically replaces what was previously being done in a custom DAO method to instead now be using the generic search DAO function to achieve the same result.